ShareTheTable/View Controllers/Profile Making/Add Photos/AddPhotosViewController.m

```diff
@@ -0,0 +1,137 @@
+    // self.imageViews = [[NSMutableArray alloc] init];
+    PHPickerConfiguration *config = [[PHPickerConfiguration alloc] init];
+    config.selectionLimit = 5;
+    config.filter = [PHPickerFilter imagesFilter];
+
+    PHPickerViewController *pickerViewController = [[PHPickerViewController alloc] initWithConfiguration:config];
+    pickerViewController.delegate = self;
+    [self presentViewController:pickerViewController animated:YES completion:nil];
+}
+
+- (void)viewDidLoad {
+    [super viewDidLoad];
+    // Do any additional setup after loading the view.
+    self.imageViews = [[NSMutableArray alloc] init];
+}
+
+/*
+#pragma mark - Navigation
+
+// In a storyboard-based application, you will often want to do a little preparation before navigation
+- (void)prepareForSegue:(UIStoryboardSegue *)segue sender:(id)sender {
+    // Get the new view controller using [segue destinationViewController].
+    // Pass the selected object to the new view controller.
+}
+*/
+
+-(UIImageView*)newImageViewForImage:(UIImage*)image {
+    UIImageView *imageView = [[UIImageView alloc] init];
+    imageView.contentMode = UIViewContentModeScaleAspectFit;
+    imageView.backgroundColor = [UIColor blackColor];
+    imageView.image = image;
+    return imageView;
+}
+
+- (UIImage *)resizeImage:(UIImage *)image withSize:(CGSize)size {
+    UIImageView *resizeImageView = [[UIImageView alloc] initWithFrame:CGRectMake(0, 0, size.width, size.height)];
+    
+    resizeImageView.contentMode = UIViewContentModeScaleAspectFill;
+    resizeImageView.image = image;
+    
+    UIGraphicsBeginImageContext(size);
+    [resizeImageView.layer renderInContext:UIGraphicsGetCurrentContext()];
+    UIImage *newImage = UIGraphicsGetImageFromCurrentImageContext();
+    UIGraphicsEndImageContext();
+    
+    return newImage;
+}
+
+- (void)picker:(nonnull PHPickerViewController *)picker didFinishPicking:(nonnull NSArray<PHPickerResult *> *)results {
+    
+    [picker dismissViewControllerAnimated:YES completion:nil];
+
+    for (PHPickerResult *result in results) {
+        NSLog(@"result: %@", result);
+        NSLog(@"%@", result.assetIdentifier);
+        NSLog(@"%@", result.itemProvider);
+        
+        // Get UIImage
+        [result.itemProvider loadObjectOfClass:[UIImage class] completionHandler:^(__kindof id<NSItemProviderReading>  _Nullable object, NSError * _Nullable error) {
+
+            NSLog(@"object: %@, error: %@", object, error);
+            if ([object isKindOfClass:[UIImage class]]) {
+                dispatch_async(dispatch_get_main_queue(), ^{
+                    //UIImageView *imageView = [self newImageViewForImage:(UIImage*)object];
+                    UIImage* imageSelected = object;
+                    imageSelected = [self resizeImage:imageSelected withSize:CGSizeMake(159,159)];
+                    [self.imageViews addObject:imageSelected];
+                    
+                    if(self.imageViews.count == 5) {
```

Comment:
A general practice is to not have magic numbers in your code. We call them magic numbers because it seems arbitrary i.e. it's not clear what it's purpose is.

Instead, save the number in a well-named constant e.g. EXPECTED_COUNT_OF_PHOTOS or MIN_NUM_OF_PIC_UPLOADS depending on what the number is used for. It should be clear why that number is being used either through the constant naming, or in a comment (why is 5 used and not 4 or 6).